task18

Roj-han opened the pull request.

Roj-han committed:
task18

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter4/task18/Task18.java:
```diff
@@ -55,7 +57,45 @@ public static void main(String[] args) {
+        return -1;
+    }
+
+}
```
Потеряна пустая строка в конце файла https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter4/task18/Task18.java:
```diff
@@ -1,5 +1,7 @@
 package com.walking.intensive.chapter4.task18;
 
+import java.util.Arrays;
```
Неиспользуемый импорт

Roj-han replied:
в своих проверках использовалась распечатка массива Arrays.toString(); убрано

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter4/task18/Task18.java:
```diff
@@ -55,7 +57,45 @@ public static void main(String[] args) {
+            return girlAges[end];
+        }
+
+        while (end > begin) {
```
Для чего нужен цикл? Кажется, он здесь избыточен

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter4/task18/Task18.java:
```diff
@@ -55,7 +57,45 @@ public static void main(String[] args) {
+        }
+
+        if (end - begin == 1) {
+            if (girlAges[begin] == targetAge || girlAges[end] > targetAge) {
```
тернарка?

Roj-han committed:
task18

Roj-han committed:
task18:
- убран неиспользуемый импорт Arrays;
- два if с return в блоке со строки 74 заменены на тернарную операцию;
- удалена лишняя пустая строка;
